File: src/main/java/project/rpg/items/food/FoodBase.java
package project.rpg.items.food;

import org.bukkit.entity.Player;
import org.bukkit.event.player.PlayerItemConsumeEvent;
import project.rpg.items.ItemBase;

public abstract class FoodBase extends ItemBase {

    public abstract void onConsume(PlayerItemConsumeEvent event);

    public void onConsume(PlayerItemConsumeEvent event,int hunger, int saturation) {
        Player player = event.getPlayer();
        player.setSaturation(player.getSaturation() + saturation);
        player.setFoodLevel(hunger);
    }

}

File: src/main/kotlin/project/rpg/items/food/FoodBase.kt
package project.rpg.items.food

import org.bukkit.event.player.PlayerItemConsumeEvent
import project.rpg.items.ItemBase

abstract class FoodBase : ItemBase() {
    abstract fun onConsume(event: PlayerItemConsumeEvent)

    open fun onConsume(event: PlayerItemConsumeEvent, hunger: Int, saturation: Int) {
        val player = event.player
        player.saturation = player.saturation + saturation
        player.foodLevel = hunger
    }
}
